pkg/service/service_test.go

```diff
@@ -91,6 +91,10 @@ var (
 	backends4 = []*lb.Backend{
 		lb.NewBackend(0, lb.TCP, net.ParseIP("10.0.0.4"), 8080),
 	}
+	samebackends = []*lb.Backend{
+		lb.NewBackend(0, lb.TCP, net.ParseIP("10.0.0.5"), 8080),
+		lb.NewBackend(0, lb.TCP, net.ParseIP("10.0.0.5"), 8080),
+	}
```

Comment:
please add this test:
```
	different = []*lb.Backend{
		lb.NewBackend(0, lb.UDP, net.ParseIP("10.0.0.5"), 8080),
		lb.NewBackend(0, lb.TCP, net.ParseIP("10.0.0.5"), 8080),
	}
```
this should not fail but my understanding is that the tests will fail for this case.

Reply:
@aanm 
I added  that testcase with adding logic for checking protocols